app-frontend/src/app/components/pages/project/createAnalysis/index.js

```diff
@@ -213,42 +213,47 @@ class ProjectCreateAnalysisPageController {
         if (this.creationPromise) {
             return;
         }
-        const modal = this.modalService.open({
-            component: 'rfConfirmationModal',
-            resolve: {
-                title: () => 'Create analyses',
-                content: () =>
-                    '<div class="text-center">' +
-                    'This will create new analysis for each layer you\'ve selected on' +
-                    `your project using the "${template.title}" template` +
-                    '</div>',
-                confirmText: () => 'Create analyses',
-                cancelText: () => 'Go back'
-            }
-        });
-        modal.result.then(() => {
-            const creationPromises = this.layerList.map((layer) => {
-                const layerTemplate = this.createLayerAnalysis(layer, template);
-                return this.analysisService.createAnalysis(layerTemplate);
+        let layersWithoutAOI = this.layers.filter(l => !_.get(l, 'geometry.type'));
+        if (layersWithoutAOI.length) {
+            this.aoiRequiredModal(layersWithoutAOI[0]);
```

Comment:
We should be able to multiple layer names here in a list instead of only showing an error for a single named layer. Alternatively, we could just not name the layer and say "Selected layer(s) do not have AOI's defined on them. Click the {{warning icon button}} on each layer to set an AOI"

Reply:
👍 Good point. That seems a bit more useful than how it is now.